distribution/helm/vector-agent/templates/configmap.yaml

```diff
@@ -6,11 +8,44 @@ metadata:
   labels:
     {{- include "libvector.labels" . | nindent 4 }}
 data:
+  {{- if .Values.customConfig }}
+  {{- if and .Values.customConfig $deprecatedConfig }}{{ fail "Can't use deprecated keys with customConfig" }}{{- end }}
+  vector.yaml: |
+{{ tpl (toYaml .Values.customConfig) . | indent 4 }}
+  {{- else if (not $deprecatedConfig) }}
+  vector.yaml: |
+    # Configuration for Vector.
+    # Docs: https://vector.dev/docs/
+    data_dir: "/vector-data-dir"
+    sources:
+      host_metrics:
+        type: host_metrics
+        filesystem:
+          devices:
+            excludes: ["binfmt_misc"]
+          filesystems:
+            excludes: ["binfmt_misc"]
+          mountpoints:
+            excludes: ["*/proc/sys/fs/binfmt_misc"]
+      internal_metrics:
+        type: internal_metrics
+      kubernetes_logs:
+        type: kubernetes_logs
+    sinks:
+      prometheus_sink:
+        type: prometheus_exporter
+        inputs: ["host_metrics", "internal_metrics"]
+        address: 0.0.0.0:9090
```

Comment:
Can we move this into a separate TOML file, and [inject it into the template](https://helm.sh/docs/chart_template_guide/accessing_files/)? This'll make it a bit easier to maintain the config in the future (syntax highlighting, editor integration, etc)

Reply:
Only that it wasn't a part of the original RFC. I don't expect the default configuration to grow/change/be notably complex so I'm not particularly concerned by the maintenance burden.

Using a separate file isn't unreasonable though, it does add some misdirection on viewing the default config though. It would make it easier to run a `vector validate`, but wouldn't match how a user would need to set up that `validate` flow.

Reply:
Agreed, it'd be nice, but I'm not too concerned since I don't expect it to change much. If that turns out to not be the case than I'd feel more strongly about extracting it.